core/app/models/spree/reimbursement.rb

```diff
@@ -59,6 +59,20 @@ class IncompleteReimbursementError < StandardError; end
     class_attribute :reimbursement_failure_hooks
     self.reimbursement_failure_hooks = []
 
+    # The store_credit_category_proc allows to fetch a specific store credit
+    # category for a given reimbursement, the fetched category is used as the
+    # category passed to Spree::Reimbursement::Credit.default_creditable_class.
+    #
+    # @param proc A proc that will accept a single "reimbursement" argument
+    class_attribute :store_credit_category_proc
+    self.store_credit_category_proc = ->(_reimbursement) {
+      name = Spree::Deprecation.silence { Spree::StoreCreditCategory.reimbursement_category_name }
```

Comment:
I'd try to avoid this `silence` since it's not thread-safe yet, see https://github.com/rails/rails/pull/37590 which is still only in Rails master. 

This means that if another thread should emit a deprecation at the same time, it will be silenced. @cedum recently worked to a PR to remove all `silence` from our codebase here https://github.com/solidusio/solidus/pull/3753 and I'd keep going in that direction.

Reply:
Agree, the biggest issue with silence is that, being a class attribute, it might leave `silence` enabled globally after some time running in a threaded environment. Here's a dumb test I've done some time ago: https://gist.github.com/cedum/8315ebdd59781ee7fcc7f452a405542f

Reply:
Do you have any example of a non-invasive (or just decent) way of going around the call to `silence`? 
Everything I can think of is quite ugly but I'd like to fix it.

Reply:
I don't think there's a way, we should just not use deprecated code at all, maybe reviewing the approach we used in this PR.

Reply:
I tried locally, I think there are ways to avoid to use procs, like using a preference as @elia suggested at some point. But maybe the best thing to do is remove the hardcoding at all and add a field to the database in the StoreCreditCategory to indicate which one should be used for the reimbursements instead? 
At that point, we can just deprecate setting and getting those class attributes. WDYT?

Reply:
@kennyadsl I pushed another commit with some changes, if those are ok I'll squash the two commits and we're good to go ✨